packages/amplify_core/lib/src/types/datastore/hub/datstore_hub_event.dart

```diff
@@ -3,13 +3,74 @@
 
 import 'package:amplify_core/amplify_core.dart';
 
+/// {@template amplify_core.datastore_hub_event}
+/// DataStore periodically publishes state notifications onto Amplify's Hub.
+/// You can subscribe to the Hub to gain insight into the internal state of the DataStore.
+///
+/// The following example shows how to listen to DataStore Hub events:
+/// ```dart
+/// Amplify.Hub.listen(HubChannel.DataStore, (msg) {
+///  if (msg.type case DataStoreHubType.ready) {
```

Comment:
It's a good idea, but for simple enum compares `==` is better, IME. I've found a lot of places where `if case` does not provide help as to whether or not the check is correct (e.g. whether it would always fail)